PusherClient/EventEmitter.cs

```diff
@@ -16,7 +16,9 @@ public void Bind(string eventName, Action<dynamic> listener)
         {
             if(_eventListeners.ContainsKey(eventName))
             {
-                _eventListeners[eventName].Add(listener);
+                // To avoid triggering the same bind multiple times

```

Comment:
You don't need to "wait" for the connection. You have a connection "event" handler which informs you when the connection state changes.